src/assets/wise5/classroomMonitor/dataExport/dataExportController.ts

```diff
@@ -2594,27 +2600,31 @@ class DataExportController {
     return compositeId;
   }
 
-  /**
-   * Check if a component type has a specific export implemented for it.
-   * @param componentType The component type.
-   * @return Whether the component type has a specific export.
-   */
-  canExportAllRevisionsForComponentDataType(componentType: string) {
-    for (const allowedComponentType of this.availableComponentAllRevisionsDataExports) {
-      if (componentType === allowedComponentType) {
+  canExportAllRevisionsForComponent(component: any) {
+    return this.canExportForComponent(component, this.allowedComponentTypesForAllRevisions);
+  }
+
+  canExportLatestRevisionsForComponent(component: any) {
+    return this.canExportForComponent(component, this.allowedComponentTypesForLatestRevisions);
+  }
+
+  canExportForComponent(component: any, allowedComponentTypes: string[]): boolean {
+    for (const allowedComponentType of allowedComponentTypes) {
+      if (component.type === allowedComponentType) {
         return true;
       }
     }
-    return false;
+    return this.isDataSpring20201EmbeddedComponentAndCanExport(component);
   }
 
-  canExportLatestRevisionsForComponentDataType(componentType: string) {
-    for (const allowedComponentType of this.availableComponentLatestRevisionsDataExports) {
-      if (componentType === allowedComponentType) {
-        return true;
-      }
+  isDataSpring20201EmbeddedComponentAndCanExport(component: any) {
+    if (
+      component.type === 'Embedded' &&
+      component.tags != null &&
+      component.tags.includes('DataSpring20201')
+    ) {
+      return true;
     }
```

Comment:
```suggestion
  isDataSpring20201EmbeddedComponentAndCanExport(component: any): boolean {
    return component.type === 'Embedded' && component.tags != null && 
        component.tags.includes('DataSpring20201');
```
This function should always return a boolean value. Right now it can return true or void.

Reply:
Ok, fixed the function.

I think we should move all the component specific exports into their own classes in another issue. All the different export types should also be in their own routes and classes (work export, events export, notebook export, etc) and done in another issue as well.